OSDOCS-17161:(CQA) APP-2: Deployments - Attachment1 (#117196)
openshift/openshift-docs: modules/deployments-deploymentconfigs.adoc

include::snippets/deployment-config-deprecated.adoc[]

> [role="_abstract"]
> A `DeploymentConfig` object in {product-title} builds on replication controllers to manage the application deployment lifecycle. You can use it to roll out image updates, run lifecycle hooks, trigger automated deployments, and scale or roll back your application.

Reviewer: ```suggestion
[role="_abstract"]
You can use `DeploymentConfig` objects in {product-title} to roll out image updates, run lifecycle hooks, trigger automated deployments, and scale or roll back applications. A `DeploymentConfig` object builds on replication controllers to manage the application deployment lifecycle. 
```

    name: frontend
  template: { ... }
  triggers:
  - type: ConfigChange
  - imageChangeParams:
      automatic: true
      containerNames:
      - helloworld
      from:
        kind: ImageStreamTag
        name: hello-openshift:latest
    type: ImageChange
  strategy:
    type: Rolling
----

* `spec.triggers.type.ConfigChange` is a configuration change trigger that creates a new replication controller whenever changes are detected in the pod template of the deployment configuration.
* `spec.triggers.type.ImageChange` is an image change trigger that causes a new deployment to be created each time a new version of the backing image is available in the named image stream.

> * `strategy.type.Rolling` is the default strategy that makes a downtime-free transition between deployments.

Reviewer: ```suggestion
* `spec.strategy.type.Rolling` is the default strategy that makes a downtime-free transition between deployments.
```
